### PR TITLE
lightning: fix test lightning_error_summary

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33614

Problem Summary:

https://github.com/pingcap/tidb/pull/33510 skipped recording retryalbe error to checkpoint so that lightning can resume automatically. However, `utils.IsRetryableError` returns `true` for all unknown error. And not all unknown error is recoverable, such as checksum mismatch. We can check for checksum mismatch error specifically, but it is a little ugly. 

https://github.com/pingcap/tidb/blob/e15e6d035872e3ce4e115db3b63d912b358f3f04/br/pkg/utils/retry.go#L132-L136

### What is changed and how it works?
Remove the condition `utils.IsRetryableError`  for recoverable errors. 

Perhaps we should add a new function like `IsRecoverableError` that treats all unknown error as unrecoverable. It may be improved in the later. This PR is just to fix the integration test.

### Check List

Tests 

- [x] Unit test
- [x] Integration test

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/restore.go [hunk 1]
```diff
@@ -972,7 +972,7 @@ func (rc *Controller) saveStatusCheckpoint(ctx context.Context, tableName string
 	switch {
 	case err == nil:
 		break
-	case utils.IsRetryableError(err), utils.MessageIsRetryableStorageError(err.Error()), common.IsContextCanceledError(err):
+	case utils.MessageIsRetryableStorageError(err.Error()), common.IsContextCanceledError(err):
```
Comment (buchuitoudegou) [inline br/pkg/lightning/restore/restore.go:975]: I agree with the plan to add a new function `IsRecoverableError`. Removing `MessageIsRetryableStorageError` can possibly fix this issue but some recoverable errors would be emitted either.

**Review by buchuitoudegou — APPROVED** [review]

File: br/pkg/lightning/restore/restore.go (hunk 1 above)
Comment (gozssky) [reply br/pkg/lightning/restore/restore.go:975]: Yes. After this PR, like DNS error would be emitted like before. Could you help add a `IsRecoverableError` later?

Comment (buchuitoudegou) [reply br/pkg/lightning/restore/restore.go:975]: > 

Sure. I'll file a new PR to fix it.

**Review by D3Hunter — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**